File: dotnet/src/SimpleOrm/ConcurrencyException.cs
namespace SimpleOrm;

/// <summary>
/// Optimistic concurrency conflict (§7.16, code <c>CRUD-010</c>): an update or
/// delete carrying a version affected zero rows — someone else changed or removed
/// the row since it was loaded. The caller decides: reload and retry, or surface it.
/// </summary>
public sealed class ConcurrencyException(string target, string message)
    : Exception($"CRUD-010 {target}: {message}")
{
    public string Code => "CRUD-010";

    public string Target { get; } = target;
}

File: php/src/Errors/ConcurrencyException.php
<?php

declare(strict_types=1);

namespace SimpleOrm\Errors;

/**
 * Optimistic-concurrency conflict (§7.16, `CRUD-010`): an update or delete with a
 * version column affected zero rows — the caller's version is stale or the row
 * is gone.
 */
final class ConcurrencyException extends SimpleOrmException
{
    public function __construct(string $target, string $message)
    {
        parent::__construct('CRUD-010', $target, $message);
    }
}
